integration/tests/bft/signature-verification/orchestrator.go

```diff
@@ -0,0 +1,165 @@
+		authorizedEventReceivedWg:  sync.WaitGroup{},
+		attackerVNNoMsgSigning:     attackerVNNoMsgSigning,
+		attackerVNWithMsgSigning:   attackerVNWithMsgSigning,
+		victimEN:                   victimEN,
+	}
+
+	return orchestrator
+}
+
+// HandleEgressEvent implements logic of processing the outgoing (egress) events received from a corrupted node.
+func (s *SignatureValidationAttackOrchestrator) HandleEgressEvent(event *insecure.EgressEvent) error {
+	lg := s.logger.With().
+		Hex("corrupt_origin_id", logging.ID(event.CorruptOriginId)).
+		Str("channel", event.Channel.String()).
+		Str("protocol", event.Protocol.String()).
+		Uint32("target_num", event.TargetNum).
+		Strs("target_ids", logging.IDs(event.TargetIds)).
+		Str("flow_protocol_event", logging.Type(event.FlowProtocolEvent)).Logger()
+
+	err := s.orchestratorNetwork.SendEgress(event)
+	if err != nil {
+		lg.Error().Err(err).Msg("could not pass through egress event")
+		return err
+	}
+
+	lg.Info().Str("event_id", event.FlowProtocolEventID.String()).Msg("egress event passed through successfully")
+	return nil
+}
+
+// HandleIngressEvent implements logic of processing the incoming (ingress) events to a corrupt node.
+// This handler will track any unauthorized messages that are expected to be blocked by libp2p for message
+// signature validation failure. It also tracks all the authorized messages that are expected to be delivered to the node.
+func (s *SignatureValidationAttackOrchestrator) HandleIngressEvent(event *insecure.IngressEvent) error {
+	lg := s.logger.With().
+		Hex("origin_id", logging.ID(event.OriginID)).
+		Str("channel", event.Channel.String()).
+		Str("corrupt_target_id", fmt.Sprintf("%v", event.CorruptTargetID)).
+		Str("flow_protocol_event", fmt.Sprintf("%T", event.FlowProtocolEvent)).Logger()
+
+	// Track any unauthorized events that are received by the corrupted node that has disabled message signing.
+	// These messages should have been rejected by libp2p.
+	if _, ok := s.unauthorizedEvents[event.FlowProtocolEventID]; ok {
+		s.unauthorizedEventsReceived.Inc()
+		lg.Warn().Str("event_id", event.FlowProtocolEventID.String()).Msg("unauthorized ingress event received")
+	}
+
+	// track all authorized events sent during test
+	if _, ok := s.authorizedEvents[event.FlowProtocolEventID]; ok {
+		s.authorizedEventsReceived.Inc()
+		s.authorizedEventReceivedWg.Done()
+	}
+
+	err := s.orchestratorNetwork.SendIngress(event)
+	if err != nil {
+		lg.Error().Err(err).Msg("could not pass through ingress event")
+		return err
+	}
+
+	lg.Info().Str("event_id", event.FlowProtocolEventID.String()).Msg("ingress event passed through successfully")
+	return nil
+}
+
+func (s *SignatureValidationAttackOrchestrator) Register(orchestratorNetwork insecure.OrchestratorNetwork) {
+	s.orchestratorNetwork = orchestratorNetwork
+}
+
+// sendUnauthorizedMsgs publishes a number of unauthorized messages without signatures from the corrupt AN with message signing enabled to the victim AN.
```

Comment:
```suggestion
// sendUnauthorizedMsgs publishes a number of unauthorized messages without signatures from one corrupt VM to another (victim) corrupt EN. The sender is corrupt since the attacker needs to take control over what it sends. Moreover, the receiver is also corrupt as the testing framework needs to have an eye on what it receives (i.e., ingress traffic). 
```
Please also double-check my revision.